r/R/dplyr.R

```diff
@@ -434,6 +438,77 @@ build_function_list <- function(FUN) {
   )
 }
 
+arrow_r_string_replace_function <- function(FUN, max_replacements) {
+  function(pattern, replacement, x, ignore.case = FALSE, fixed = FALSE) {
+    if (ignore.case) {
+      if (fixed) {
+        pattern <- gsub("\\E", "\\e", pattern, fixed = TRUE)
+        pattern <- paste0("(?i)\\Q", pattern, "\\E")
+        replacement <- gsub("\\", "\\\\", replacement, fixed = TRUE)
+      } else {
+        pattern <- paste0("(?i)", pattern)
+      }
+    }
```

Comment:
@maartenbreddels since ARROW-10306 did not directly support string-insensitive fixed string replacement (non-regex), this is the workaround we used to support this in the R bindings. Feedback welcome, and thanks for your work on the underlying C++ code!

Reply:
Comments added explaining this code in https://github.com/apache/arrow/pull/9878/commits/db828beec72f5cac9e673bf4e3305c077e1bad5d